src/python/dxpy/bindings/dxfile.py

```diff
@@ -363,6 +396,16 @@ def write(self, data, multithread=True, **kwargs):
             does not affect where the next :meth:`write` will occur.
 
         '''
+        self.file_limits = dxpy.api.project_describe(self.project, {'fields': {'fileUploadParameters': True}})['fileUploadParameters']
+        self._set_file_limits(self.file_limits)
+
+        self._write_buf_size = self._get_buffer_size_for_file(self.file_size, self.file_is_mmapd)
```

Comment:
Wouldn't that be a change to the existing functionality?  Does this code adjust the buffer size to the defaults for max and min?

```buffer_size = max(self._buffer_size, min_buffer_size)```

So, wouldn't we basically just be removing the bit below?
```
if buffer_size * self._maximum_parts < expected_file_size:
            raise AssertionError('part size is not large enough to complete upload')
```